### DISCUSSION TITLE
Welcome to ocsf-java-tools Discussions!

### DISCUSSION THREAD

**Discussion by rroupski** [ocsf/ocsf-java-tools#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by raghavendrakumarb** [comment]
Hi,

Would like to know the maven / gradle dependency information to use ocsf-java-tools?  Did not find any release information mentioned against the repository.

Thanks

**Comment by raghavendrakumarb** [comment]
Hi @rroupski / community developers,

Need your suggestions on Translation rules using **ocsf-translator** for the below questions

**Question 1**) How to write **Translation rules** on **array of entries**?

For example would like to map entries in Source JSON interfaces array to Target JSON's interfaces array 
wherein each entry in interfaces  object has to do the following mappings : gateway_mac to gw_mac , gateway_ip to gw_ip ?
Please suggest any Example available in the code/documentation to do such translations. ?


Source JSON:

```
{
	"source_network_information": {
		"interfaces": [{
				"name": "n123",
				"gateway_mac": "ab: 44: 50: 8 f",
				"gateway_ip": "192.1 .2 .3",
				"connectivity": "ethernet0"
			},
			{
				"name": "n124",
				"gateway_mac": "ab: 44: 50: 12",
				"gateway_ip": "192.168 .1 .1",
				"connectivity": "ethernet1"
			}
		]
	}
}
```



Target JSON:

```
{
	"target_network_information": {
		"interfaces": [{
			"name": "n123",
			"gw_mac": "ab:44:50:8f",
			"gw_ip": "192.1.2.3",
			"connectivity": {
				"name": "ethernet0"
			}
		}, {
			"name": "n124",
			"gw_mac": "ab: 44: 50: 12",
			"gw_ip": "192.168 .1 .1",
			"connectivity": {
				"name": "ethernet1"
			}
		}]
	}
}
```

**Question 2** ) write custom java code / specify java method call in the translation rules definition ?
